packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart

```diff
@@ -116,7 +116,7 @@ class ScrollPositionWithSingleContext extends ScrollPosition implements ScrollAc
     super.beginActivity(newActivity);
     _currentDrag?.dispose();
     _currentDrag = null;
-    if (!activity.isScrolling)
+    if (activity is IdleScrollActivity)
```

Comment:
```suggestion
    if (!activity.isScrolling && activity is! HoldScrollActivity)
```
May still be better.. I have not found a corner case that proves this to be bad. And all existing tests have passed with either implementation.